### PR TITLE
AEA-1575 add tests on MAM with password

### LINKED ISSUE #2471: Missing tests for password usage
**Is your feature request related to a problem? Please describe.**
We should add tests for password usage in CLI and manager

**Describe the solution you'd like**
Obvs.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

add tests on MAM with password

## Fixes

Fix #2471

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/test_tools/test_cases.py [hunk 1]
```diff
@@ -661,18 +673,26 @@ def replace_private_key_in_file(
...
+    def generate_wealth(
+        cls, ledger_api_id: str = DEFAULT_LEDGER, password: Optional[str] = None
+    ) -> Result:
         """
         Generate wealth with CLI command.
 
         Run from agent's directory.
 
         :param ledger_api_id: ledger API ID.
+        :param password: the password option.
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:685]:
```suggestion
        :param password: the password.
```

**Review by DavidMinarsch — APPROVED** [review]
LGTM, just minor docstring issues need fixing

**Review by solarw — APPROVED** [review]
LGTM

File: tests/test_cli/test_get_address.py [hunk 2]
```diff
@@ -67,3 +68,43 @@ def test_run_positive(self, *mocks):
...
+        """Tear down the test."""
+        super().teardown_class()
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """
+        Override the 'teardown_class' method.
+
+        This will prevent teardown of tests at class-level.
+        """
```
Comment (marcofavorito) [inline tests/test_cli/test_get_address.py:110]: @DavidMinarsch @solarw this is a nice pattern we could support in the AEA testing framework. `setup` and `teardown` are called before/after each test; maybe a bit inefficient (a temporary AEA project has to be created each time), but useful for e.g. parameterization as in this case.

Comment (DavidMinarsch) [reply tests/test_cli/test_get_address.py:110]: I think its ok to do here, but if we were to do it on the base class we would conflate the meaning of setup and setup_class. I am not sure that this is desirable.

**Pull request merged**